react/features/base/tracks/middleware.any.ts

```diff
@@ -210,11 +211,16 @@ async function _setMuted(store: IStore, { ensureTrack, authority, muted }: {
         if (jitsiTrack && (
             jitsiTrack.videoType !== 'desktop' || isAudioOnly || getMultipleVideoSendingSupportFeatureFlag(state))
         ) {
-            setTrackMuted(jitsiTrack, muted, state).catch(() => dispatch(trackMuteUnmuteFailed(localTrack, muted)));
+            setTrackMuted(jitsiTrack, muted, state, dispatch)
+                .catch(() => dispatch(trackMuteUnmuteFailed(localTrack, muted)));
         }
     } else if (!muted && ensureTrack && (typeof APP === 'undefined' || isPrejoinPageVisible(state))) {
+        typeof APP !== 'undefined' && dispatch(gumPending([ mediaType ], IGUMPendingState.PENDING_UNMUTE));
+
         // FIXME: This only runs on mobile now because web has its own way of
         // creating local tracks. Adjust the check once they are unified.
-        dispatch(createLocalTracksA({ devices: [ mediaType ] }));
+        dispatch(createLocalTracksA({ devices: [ mediaType ] })).then(() => {
```

Comment:
This happens only on mobile as the comment suggest, so we are inadvertently setting the gUM pending state on mobile, is it that we want ?

Reply:
To be honest I haven't paid attention to the comment but actually in the `if` clause you can see that it can be executed for the prejoin screen. I guess the comment is just outdated...